app/controllers/provider_orders_controller.rb

```diff
@@ -131,4 +151,19 @@ def generate_provider_order(guid)
       { 'error' => error_message }
     end
   end
+
+  # sets up initial values to track time spent issuing faraday requests.
+  # initial budget of time allowed to complete request is 270 seconds (300-30 see below)
+  # echo_client.timeout as of 5/16/19 is 300 seconds, subtracting 30 seconds for any potential processing,
+  # the rest of the remaining time will be used for faraday requests.
+  def init_time_tracking_variables
```

Comment:
you're declaring these time helper methods twice. it might be better to put it higher in the inheritance chain (maybe `manage_cmr_controller` so both controllers can access them. if you do that, explaining that the time tracking is for orders would be helpful.

Reply:
I was one the fence about this one, as usually bad idea to have redundant code.   Agree, I moved into manage_cmr_controller.